Source/Details/_ASDisplayViewAccessiblity.mm

```diff
@@ -109,64 +121,112 @@ - (CGRect)accessibilityFrame
+                 ASIsLeafNode(currentNode) &&
+                 currentNode.accessibilityElementCount > 0) {
+        // In leaf nodes that are layer backed and acting as UIAccessibilityContainer
+        // (isAccessibilityElement == NO we call through to the
+        // accessibilityElements to collect all accessibility elements of this node
+        [elements addObjectsFromArray:currentNode.accessibilityElements];
+      }
     }
   });
 }
 
-static void CollectAccessibilityElementsForContainer(ASDisplayNode *container, UIView *view,
-                                                     NSMutableArray *elements) {
-  ASDisplayNodeCAssertNotNil(view, @"Passed in view should not be nil");
-  if (view == nil) {
+///// Called from CollectAccessibilityElements for nodes that are returning YES for isAccessibilityContainer to collect all subnodes accessibility labels as well as custom actions for nodes that have interactive accessibility traits enabled. Furthermore for ASTextNode's it also aggregates all links within the attributedString as custom action
+static void AggregateSublabelsOrCustomActionsForContainerNode(ASDisplayNode *containerNode, UIView *containerView, NSMutableArray *elements) {
+  ASDisplayNodeCAssertNotNil(containerView, @"Passed in view should not be nil");
+  if (containerView == nil) {
     return;
   }
   UIAccessibilityElement *accessiblityElement =
-      [ASAccessibilityElement accessibilityElementWithContainer:view
-                                                           node:container];
+      [ASAccessibilityElement accessibilityElementWithContainerView:containerView node:containerNode];
 
   NSMutableArray<ASAccessibilityElement *> *labeledNodes = [[NSMutableArray alloc] init];
   NSMutableArray<ASAccessibilityCustomAction *> *actions = [[NSMutableArray alloc] init];
   std::queue<ASDisplayNode *> queue;
-  queue.push(container);
+  queue.push(containerNode);
 
   // If the container does not have an accessibility label set, or if the label is meant for custom
   // actions only, then aggregate its subnodes' labels. Otherwise, treat the label as an overriden
   // value and do not perform the aggregation.
   BOOL shouldAggregateSubnodeLabels =
-      (container.accessibilityLabel.length == 0) ||
-      (container.accessibilityTraits & ASInteractiveAccessibilityTraitsMask());
+      (containerNode.accessibilityLabel.length == 0) ||
+      (containerNode.accessibilityTraits & ASInteractiveAccessibilityTraitsMask());
 
+  // Iterate through the whole subnode tree and aggregate
   ASDisplayNode *node = nil;
   while (!queue.empty()) {
     node = queue.front();
     queue.pop();
 
-    if (node != container && node.isAccessibilityContainer) {
-      UIView *containerView = node.isLayerBacked ? view : node.view;
-      CollectAccessibilityElementsForContainer(node, containerView, elements);
+    // If the node is an accessibility container go further down for collecting all the nodes
+    // information
```

Comment:
Add period to the end of these comments. Possibly use a longer line length.

Reply:
Done